### ISSUE TITLE
can't pickle NgramModels because we can't pickle ConditionalFreqDists because Python can't pickle lambdas ...

### ISSUE THREAD

**Issue by alexrudnick** [nltk/nltk#393]
Right now, we can't pickle an NgramModel, like so:

``` python
lm = nltk.model.ngram.NgramModel(3, ['a', 'b', 'c'])
pickle.dumps(lm)
```

... but I know why this is! We can't pickle any ConditionalFreqDist, and the NgramModel has one of those as a member.

``` python
from nltk.probability import ConditionalProbDist
from nltk.probability import ConditionalFreqDist
from nltk.probability import SimpleGoodTuringProbDist
cfd = ConditionalFreqDist()
cfd['hello'].inc('everybody')
cfd['hello'].inc('everybody')
model = ConditionalProbDist(cfd, SimpleGoodTuringProbDist)
pickle.dumps(model)
```

And this can't be done because ConditionalProbDist carries a lambda around.
https://github.com/nltk/nltk/blob/master/nltk/probability.py#L2091

Any thoughts on the best way to fix this, so we can pickle NgramModels again? I'll take a stab at it...

**Comment by kmike** [comment]
I think the easiest way to fix ConditionalProbDist pickling is to uncomment code from the constructor and write `__reduce__()`.

**Comment by kmike** [comment]
Alex, is NgramModel picklable now?

**Comment by alexrudnick** [comment]
It is! We should probably add a test to make sure it stays picklable.

I suppose we should start a file for NgramModel tests.

I'll go ahead and close this issue, though.

**Comment by alexrudnick** [comment]
Fixed by pull #394.
